src/TulsiGenerator/Scripts/bazel_build.py

```diff
@@ -608,6 +613,12 @@ def Run(self, args):
       # the host itself.
       if (self.is_test and self.xcode_version_minor >= 730 and
           self.codesigning_allowed):
+        # If we're codesigning we also need to make sure we use the correct
+        # provisioning profile.
+        if not self.is_simulator and self.test_runner_provisioning_profile:
```

Comment:
You can remove the check for `self.is_simulator`. The value `self.codesigning_allowed` includes this check baked.